stock-service/src/main/java/com/ecommercepractice/stockservice/controller/StockController.java

```diff
@@ -0,0 +1,106 @@
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.hateoas.EntityModel;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import javax.validation.Valid;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Controller
+@RequestMapping("/api/v1/stock")
+public class StockController {
+    @Autowired
+    StockService stockService;
+
+    @Autowired
+    StockModelAssembler stockModelAssembler;
+
+    @ApiOperation(value = "APPEND_TO_STOCK", notes = "Append a product to stock")
+    @PostMapping()
+    public ResponseEntity<EntityModel<Stock>> appendToInventory(@Valid @RequestBody Stock productStock) {
+        log.info(String.format("STOCK | APPEND | PAYLOAD { Stock -> %s } ", productStock));
+        return new ResponseEntity(stockModelAssembler.toModel(stockService.appendToInventory(productStock)), HttpStatus.CREATED);
+    }
+
+    @ApiOperation(value ="FILTER", notes = "Fetch products by filtering them by price, stockQuantity and soldQuiantity" +
+            "Also, you can specify whether you want to list not onlyProducts also")
+    @GetMapping()
+    public ResponseEntity<List<EntityModel<Stock>>>fetchByFiltering(@RequestParam(required = false) Float price,
+                                                        @RequestParam(required = false) Long stockQuantity,
+                                                        @RequestParam(required = false) Long soldQuantity,
+                                                        @RequestParam(required = false, defaultValue = "true") Boolean onlyActive) {
+        log.info(String.format("STOCK | FETCH | STOCK NAME -> { %s } PRICE -> { %s }  ",
+                price == null ? "NO_PRICE" : price.toString(),
+                stockQuantity == null ? "NO_stockQuantity" : stockQuantity.toString(),
+                soldQuantity == null ? "NO_soldQuantity" : soldQuantity.toString()));
+
+        List<EntityModel<Stock>> stockList = stockService.fetchByFiltering(price, stockQuantity, soldQuantity, onlyActive)
+                .stream()
+                .map(stockModelAssembler::toModel)
+                .collect(Collectors.toList());
+
+        return new ResponseEntity(stockList, HttpStatus.OK);
+    }
+
+    @ApiOperation(value ="FETCH BY ID", notes = "Get a product on stock by their id.")
+    @GetMapping("/{stockId}")
```

Comment:
please, try to set a resource more clear for example /stock/products/{product-id}

Reply:
Done